packages/google_maps_flutter/google_maps_flutter/ios/Classes/FLTGoogleMapTileOverlayController.m

```diff
@@ -93,22 +65,53 @@ - (void)setZIndex:(int)zIndex {
 - (void)setTileSize:(NSInteger)tileSize {
   self.layer.tileSize = tileSize;
 }
+
+- (void)interpretTileOverlayOptions:(NSDictionary *)data {
+  if (!data) {
+    return;
+  }
+  NSNumber *visible = data[@"visible"];
+  if (visible != nil && visible != (id)[NSNull null]) {
+    [self setVisible:visible.boolValue];
+  }
+
+  NSNumber *transparency = data[@"transparency"];
+  if (transparency != nil && transparency != (id)[NSNull null]) {
+    [self setTransparency:transparency.floatValue];
+  }
+
+  NSNumber *zIndex = data[@"zIndex"];
+  if (zIndex != nil && zIndex != (id)[NSNull null]) {
+    [self setZIndex:zIndex.intValue];
+  }
+
+  NSNumber *fadeIn = data[@"fadeIn"];
+  if (fadeIn != nil && fadeIn != (id)[NSNull null]) {
+    [self setFadeIn:fadeIn.boolValue];
+  }
+
+  NSNumber *tileSize = data[@"tileSize"];
+  if (tileSize != nil && tileSize != (id)[NSNull null]) {
+    [self setTileSize:tileSize.integerValue];
+  }
+}
+
 @end
 
 @interface FLTTileProviderController ()
 
-@property(weak, nonatomic) FlutterMethodChannel *methodChannel;
```

Comment:
Do you remember why you made this weak in the first place?
https://github.com/flutter/plugins/pull/3434/files#diff-7013871fe3ee02101b28aad71693b5ef3d5f48414c744ddef944e57fe886c5f4R100

Reply:
Then did you mean to make it strong in this PR?

Reply:
Yep, to match other controllers.

Reply:
So should they all be `weak` as you had it before? I'm not trying to argue one way or the other.

Reply:
I think Strong is better. There won't be a retain cycle as method channel is not going to have a strong reference to the controllers. 
Having them as weak makes the methodChannel's life cycle confusing. One needs to figure out who strongly owns the `methodChannel` (GoogleMapsController) to understand its life cycle.